ctapipe/tools/stage1.py

```diff
@@ -0,0 +1,731 @@
+        help="compression level, 0=None, 9=maximum", default_value=5, min=0, max=9
+    ).tag(config=True)
+
+    split_datasets_by = CaselessStrEnum(
+        values=["tel_id", "tel_type"],
+        default_value="tel_id",
+        help="Splitting level for the parameters and images datasets",
+    ).tag(config=True)
+
+    compression_type = CaselessStrEnum(
+        values=["blosc:zstd", "zlib"],
+        help="compressor algorithm to use. ",
+        default_value="zlib",
+    ).tag(config=True)
+
+    image_extractor_type = create_class_enum_trait(
+        base_class=ImageExtractor,
+        default_value="NeighborPeakWindowSum",
+        help="Method to use to turn a waveform into a single charge value",
+    ).tag(config=True)
+
+    gain_selector_type = create_class_enum_trait(
+        base_class=GainSelector, default_value="ThresholdGainSelector"
+    ).tag(config=True)
+
+    image_cleaner_type = create_class_enum_trait(
+        base_class=ImageCleaner, default_value="TailcutsImageCleaner"
+    )
+
+    write_index_tables = Bool(
+        help=(
+            "Generate PyTables index datasets for all tables that contain an "
+            "event_id or tel_id. These speed up in-kernal pytables operations,"
+            "but add some overhead to the file. They can also be generated "
+            "and attached after the file is written "
+        ),
+        default_value=False,
+    ).tag(config=True)
+
+    overwrite = Bool(help="overwrite output file if it exists").tag(config=True)
+    progress_bar = Bool(help="show progress bar during processing").tag(config=True)
+
+    aliases = {
+        "input": "EventSource.input_url",
+        "output": "Stage1ProcessorTool.output_path",
+        "allowed-tels": "EventSource.allowed_tels",
+        "max-events": "EventSource.max_events",
+        "image-extractor-type": "Stage1ProcessorTool.image_extractor_type",
+        "gain-selector-type": "Stage1ProcessorTool.gain_selector_type",
+        "image-cleaner-type": "Stage1ProcessorTool.image_cleaner_type",
+    }
+
+    flags = {
+        "write-images": (
+            {"Stage1ProcessorTool": {"write_images": True}},
+            "store DL1/Event/Telescope images in output",
+        ),
+        "write-parameters": (
+            {"Stage1ProcessorTool": {"write_parameters": True}},
+            "store DL1/Event/Telescope parameters in output",
+        ),
+        "write-index-tables": (
+            {"Stage1ProcessorTool": {"write_index_tables": True}},
+            "generate PyTables index tables for the parameter and image datasets",
+        ),
+        "overwrite": (
+            {"Stage1ProcessorTool": {"overwrite": True}},
+            "Overwrite output file if it exists",
+        ),
+        "progress": (
+            {"Stage1ProcessorTool": {"progress_bar": True}},
+            "show a progress bar during event processing",
+        ),
+    }
+
+    classes = List(
+        [EventSource, CameraCalibrator, ImageQualityQuery]
```

Comment:
I think this should be `classes_with_traits(EventSource)` since one needs to configure the event source class one is actually using. Setting options for the `EventClass` does not work for subclasses.

Reply:
yes, done.